src/solvers/dgsem_p4est/dg_3d_parabolic.jl

```diff
@@ -0,0 +1,717 @@
+                                                                                       m,
+                                                                                       element] +
+                                                                        surface_flux_values[v,
+                                                                                            l,
+                                                                                            m,
+                                                                                            4,
+                                                                                            element] *
+                                                                        factor_2 *
+                                                                        normal_direction[dim])
+
+                        # surface at -z
+                        normal_direction = get_normal_direction(5, contravariant_vectors,
+                                                                l, m, 1, element)
+                        gradients[dim][v, l, m, 1, element] = (gradients[dim][v, l, m, 1,
+                                                                              element] +
+                                                               surface_flux_values[v, l, m,
+                                                                                   5,
+                                                                                   element] *
+                                                               factor_1 *
+                                                               normal_direction[dim])
+
+                        # surface at +z
+                        normal_direction = get_normal_direction(6, contravariant_vectors,
+                                                                l, m, nnodes(dg), element)
+                        gradients[dim][v, l, m, nnodes(dg), element] = (gradients[dim][v, l,
+                                                                                       m,
+                                                                                       nnodes(dg),
+                                                                                       element] +
+                                                                        surface_flux_values[v,
+                                                                                            l,
+                                                                                            m,
+                                                                                            6,
+                                                                                            element] *
+                                                                        factor_2 *
+                                                                        normal_direction[dim])
```

Comment:
I think this entire function either needs to be refactored to be legible or else it should be wrapped in
```julia
#! format: off
...
#! format: on
```
and hand-formatted to be legible again.

Reply:
I had hoped that these formatting related discussions are a thing of the past, but I am starting to get a bad feeling... 😢

Reply:
Something like setting `grad = gradients[dim]` and using it below should already help?

Reply:
Thank you for the reviews. I applied Ranocha's suggestion and reformatted the loop. Can you please take a look now? @ranocha @sloede

Reply:
Looks much better now, thanks